test/unistd/io.c

```diff
@@ -63,8 +61,11 @@ int main() {
 
     // NB: These are meant to test FS.createDevice specifically,
     //     and as such do not use registerDevice/mkdev
+#if !defined(WASMFS)
+    // WasmFS does not support FS.createDevice()
```

Comment:
There's no good reason now that I think about it. I think earlier we discussed that since `FS.createDevice` was only used internally and it seemed difficult to match the old API, it might be better to skip. However, some of the other changes I've made since then have made it a little easier to implement `createDevice`, so I've added it to the PR.